### ISSUE TITLE
Build fails. v1.14.1, rust 1.80.0

### ISSUE THREAD

**Issue by erik78se** [paritytech/polkadot-sdk#5164]
I was trying to build polkadot v1.14.1 with rust 1.80.0, Linux x86_64, ubuntu22.04

What am I doing wrong?

```
2024-07-27 16:29:07.968 :: 2024-07-27 16:28:44.168 ::    Compiling proc-macro-warning v1.0.0
2024-07-27 16:29:07.968 :: 2024-07-27 16:28:44.664 :: error[E0282]: type annotations needed for `Box<_>`
2024-07-27 16:29:07.968 :: 2024-07-27 16:28:44.664 ::   --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/time-0.3.27/src/format_description/parse/mod.rs:83:9
2024-07-27 16:29:07.968 :: 2024-07-27 16:28:44.664 ::    |
2024-07-27 16:29:07.968 :: 2024-07-27 16:28:44.664 :: 83 |     let items = format_items
2024-07-27 16:29:07.968 :: 2024-07-27 16:28:44.664 ::    |         ^^^^^
2024-07-27 16:29:07.968 :: 2024-07-27 16:28:44.664 :: ...
2024-07-27 16:29:07.968 :: 2024-07-27 16:28:44.665 :: 86 |     Ok(items.into())
2024-07-27 16:29:07.968 :: 2024-07-27 16:28:44.665 ::    |              ---- type must be known at this point
2024-07-27 16:29:07.968 :: 2024-07-27 16:28:44.665 ::    |
2024-07-27 16:29:07.968 :: 2024-07-27 16:28:44.665 :: help: consider giving `items` an explicit type, where the placeholders `_` are specified
2024-07-27 16:29:07.968 :: 2024-07-27 16:28:44.665 ::    |
2024-07-27 16:29:07.968 :: 2024-07-27 16:28:44.665 :: 83 |     let items: Box<_> = format_items
2024-07-27 16:29:07.968 :: 2024-07-27 16:28:44.665 ::    |              ++++++++
2024-07-27 16:29:07.968 :: 2024-07-27 16:28:44.665 ::
2024-07-27 16:29:07.968 :: 2024-07-27 16:28:44.707 ::    Compiling proc-macro-crate v1.3.1
2024-07-27 16:29:07.968 :: 2024-07-27 16:28:44.793 ::    Compiling anstyle v1.0.6
2024-07-27 16:29:07.968 :: 2024-07-27 16:28:45.209 ::    Compiling nom v7.1.3
2024-07-27 16:29:07.968 :: 2024-07-27 16:28:45.306 ::    Compiling polyval v0.6.1
2024-07-27 16:29:07.968 :: 2024-07-27 16:28:45.358 ::    Compiling openssl-sys v0.9.102
2024-07-27 16:29:07.968 :: 2024-07-27 16:28:45.871 :: For more information about this error, try `rustc --explain E0282`.
2024-07-27 16:29:07.968 :: 2024-07-27 16:28:45.885 :: error: could not compile `time` (lib) due to 1 previous error
2024-07-27 16:29:07.968 :: 2024-07-27 16:28:45.886 :: warning: build failed, waiting for other jobs to finish...
2024-07-27 16:29:07.968 :: 2024-07-27 16:29:07.508 :: error: failed to compile `polkadot v6.0.0 (/root/parts/polkadot/build/polkadot)`, intermediate artifacts can be found at `/root/parts/polkadot/build/target`.

```

**Comment by gui1117** [comment]
The error comes from `time` crate. Update it:
```
cargo update -p time
```

**Comment by erik78se** [comment]
I can confirm that it builds with the suggested fix.

Would this be something that needs to go into a pinned version for the build?

**Comment by bkchr** [comment]
The release is mentioning which Rustc was used to compile it: https://github.com/paritytech/polkadot-sdk/releases/tag/polkadot-v1.14.1

Best to use this version. Future versions will fix the build with rustc 1.80.0.

**Comment by erik78se** [comment]
> The release is mentioning which Rustc was used to compile it: https://github.com/paritytech/polkadot-sdk/releases/tag/polkadot-v1.14.1
> 
> Best to use this version. Future versions will fix the build with rustc 1.80.0.

Totally - with the simple update of time, I managed to build, run and test with rustc 1.80 but yeah, hopefully its fixed in next release.

**Comment by gituser** [comment]
why not update `time` crate in the latest release?

thanks @erik78se for creating this issue it helped me too to compile latest polkadot release

P.S. I think some ppl always update rustc before compiling polkadot
